src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +230,213 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+  private Future<HttpResponse<Buffer>> logout(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ENDPOINT);
+
     request.putHeader(XOkapiHeaders.TENANT, tenant)
-      .putHeader(XOkapiHeaders.TOKEN, requestToken);
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
 
-    JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    return request
-      .expect(ResponsePredicate.SC_CREATED)
-      .expect(ResponsePredicate.JSON)
-      .sendJsonObject(payload)
-      .map(response -> response.bodyAsJsonObject().getString("refreshToken")
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    logger.debug("Cookie expires is {}", ttlSeconds);
+    return Cookie.cookie(REFRESH_TOKEN, refreshToken)
+        .setMaxAge(ttlSeconds)
+        .setSecure(true)
+        .setPath("/authn")
+        .setHttpOnly(true)
+        .encode();
+  }
+
+  private String accessTokenCookie(String accessToken, String accessTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(accessTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    return Cookie.cookie(ACCESS_TOKEN, accessToken)
+      .setMaxAge(ttlSeconds)
+      .setSecure(true)
+      .setHttpOnly(true)
+      .encode();
```

Comment:
You might add a comment saying

    // Don't set Domain so that browsers send cookie to Okapi host only, not including subdomains.
    // Setting Domain will send cookie to subdomains.
    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value